apps/web/src/components/WalletConnect/WalletConnectProvider.tsx

```diff
@@ -0,0 +1,126 @@
+          `The requested required network "${network}" is not supported. Available: ${availablenetworks}`
+        );
+      }
+
+      await openWith(<SessionProposalModal network={network} proposal={proposal} />, {});
+      console.log("Session proposal from dApp", proposal, walletKit.getActiveSessions());
+    }).catch(async () => {
+      // dApp is waiting so we need to notify it
+      await walletKit.rejectSession({ id: proposal.id, reason: getSdkError("UNSUPPORTED_CHAINS") });
+    });
+
+  const onSessionDelete = (event: WalletKitTypes.SessionDelete) => {
+    // by that time the session is already deleted from WalletKit so we cannot find the dApp name
+    console.log("Session deleted by dApp", event);
+    toast({
+      description: "Session deleted by peer dApp",
+      status: "info",
+    });
+  };
+
+  const onSessionRequest = async (event: WalletKitTypes.SessionRequest) => {
+    try {
+      const activeSessions: Record<string, SessionTypes.Struct> = walletKit.getActiveSessions();
+      if (!(event.topic in activeSessions)) {
+        console.error("WalletConnect session request failed. Session not found", event);
+        throw new Error("WalletConnect session request failed. Session not found");
+      }
+
+      const session = activeSessions[event.topic];
+
+      console.log("Session request, event, session", event, session);
+      console.log("Session request from dApp", session);
+      toast({
+        description: `Session request from dApp ${session.peer.metadata.name}`,
+        status: "info",
+      });
+      throw new Error("Not implemented");
+    } catch (error) {
+      const { id, topic } = event;
+      const activeSessions: Record<string, SessionTypes.Struct> = walletKit.getActiveSessions();
+      console.error("WalletConnect session request failed", event, error);
+      if (event.topic in activeSessions) {
+        const session = activeSessions[event.topic];
+        toast({
+          description: `Session request for dApp ${session.peer.metadata.name} failed. It was rejected.`,
+          status: "error",
+        });
+      } else {
+        toast({
+          description: `Session request for dApp ${topic} failed. It was rejected. Peer not found by topic.`,
+          status: "error",
+        });
+      }
+      // dApp is waiting so we need to notify it
+      const response = formatJsonRpcError(id, getSdkError("INVALID_METHOD").message);
+      await walletKit.respondSessionRequest({ topic, response });
+    }
+  };
+
+  useEffect(() => {
+    const initializeWallet = async () => {
+      // create the waalet just once
+      await createWalletKit();
+      walletKit.on("session_proposal", event => void onSessionProposal(event));
+      walletKit.on("session_request", event => void onSessionRequest(event));
+      walletKit.on("session_delete", event => void onSessionDelete(event));
+    };
+    void initializeWallet();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
as I said before, please don't disable this check unless you **really** understand what you're doing